operator-framework-core/src/main/java/io/javaoperatorsdk/operator/Operator.java

```diff
@@ -104,9 +157,10 @@ public void close() {
    * @param <R> the {@code CustomResource} type associated with the controller
    * @throws OperatorException if a problem occurred during the registration process
    */
-  public <R extends CustomResource> void register(
+  private <R extends CustomResource> void startController(
       ResourceController<R> controller, ControllerConfiguration<R> configuration)
       throws OperatorException {
+
     final var existing = configurationService.getConfigurationFor(controller);
     if (existing == null) {
```

Comment:
@metacosm @csviri Not strictly related to this PR but is this check correct ? I'd expect this condition to be evaluated only if the given `configuration` parameter is null.

Reply:
@lburgazzoli this check is correct: the intent is that the provided configuration is an override of the existing configuration. To that end, the controller must already been known to the configuration service, which is why we first check if we have an existing configuration which is used if no overridden version is provided.